api/webhook.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2015 Spinpunch, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package api
+
+import (
+	l4g "code.google.com/p/log4go"
+	"github.com/gorilla/mux"
+	"github.com/mattermost/platform/model"
+	"github.com/mattermost/platform/utils"
+	"net/http"
+)
+
+func InitWebhook(r *mux.Router) {
+	l4g.Debug("Initializing webhook api routes")
+
+	sr := r.PathPrefix("/hooks").Subrouter()
+	sr.Handle("/incoming/create", ApiUserRequired(createIncomingHook)).Methods("POST")
+	sr.Handle("/incoming/delete", ApiUserRequired(deleteIncomingHook)).Methods("POST")
+	sr.Handle("/incoming/list", ApiUserRequired(getIncomingHooks)).Methods("GET")
+}
+
+func createIncomingHook(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !utils.Cfg.ServiceSettings.AllowIncomingWebhooks {
+		c.Err = model.NewAppError("createIncomingHook", "Incoming webhooks have been disabled by the system admin.", "")
+		c.Err.StatusCode = http.StatusNotImplemented
+		return
+	}
+
+	c.LogAudit("attempt")
```

Comment:
Intended. We log an audit when we receive an attempt to create an incoming webhook since once the webhook is created, whoever has it can post as that user. We do audit logs like this in multiple places as well (usually for password related or user modifying related stuff)